atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyDnsMetrics.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.cluster.messaging.impl;
+
+import io.netty.channel.ChannelFuture;
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsResponseCode;
+import io.netty.resolver.dns.DnsQueryLifecycleObserver;
+import io.prometheus.client.Counter;
+import java.net.InetSocketAddress;
+import java.util.List;
+
+final class NettyDnsMetrics implements DnsQueryLifecycleObserver {
```

Comment:
👍🏼 

Was first to write lets remove the parameters, but then realized we use here the interface and register this.